examples/build.py

```diff
@@ -101,7 +101,7 @@ def exec_and_redirect(args, env=None):
 
 for plan in PLANS:
     exec_and_redirect(["./test_mmtk"], env={LIBRARY_PATH: ".", "MMTK_PLAN": plan})
-    if system == "Linux":
+    if system == "Linux" and plan != 'PageProtect': # FIXME: PageProtect for 32bit MMTk does not work yet.
```

Comment:
@javadamiri This is what I was talking about. This test basically just does a few allocation - there should be some obvious issues if it does not work. It would be good if you can help figure out what happens here. If this is not a bug in PageProtect but rather an existing bug, we can leave this alone, and fix it later. We have JikesRVM (32bit) with large object allocation, so large object allocation on 32 bit was constantly tested. So I am not sure if this is an existing issue.

Reply:
The reason page protect fails in this test is an inconsistency between `CommonFreeListPageResource::start` which is initialised to `AVAILABLE_START` and `HeapMeta.heap_cursor` which is initialised to `HEAP_START`.
In 64-bits targets, both of these are set to `HEAP_START`.

I'm still checking why this didn't cause any issue in mmtk-jikesrvm before, but I don't understand how vm space of jikesrvm is handled by mmtk-core.

Reply:
VM space is created and mmapped by mmtk-core. It is part of our heap. Is the difference caused by the page resource being contiguous for 64 bit and discontiguous for 32 bit?

Reply:
I tested several benchmarks with JikesRVM and none of them call LOS::alloc.
However, I may be doing something wrong.
How were we continuously testing LOS?

Reply:
How did you run the benchmarks? This is what I got. I put a `info!` in `LargeObjectAllocator.alloc()`.
```
$ dist/RBaseBaseSemiSpace_x86_64_m32-linux/rvm -Xms75M -Xmx75M -jar /usr/share/benchmarks/dacapo/dacapo-2006-10-MR2.jar fop
[2021-06-24T00:05:02Z INFO  mmtk::memory_manager] Initialized MMTk with SemiSpace
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(274444)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(274444)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(90124)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(8204)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(274444)
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(8204)
===== DaCapo fop starting =====
[2021-06-24T00:05:02Z INFO  mmtk::plan::global]   [POLL] copyspace0: Triggering collection
[2021-06-24T00:05:02Z INFO  mmtk::scheduler::gc_work] End of GC
[2021-06-24T00:05:02Z INFO  mmtk::plan::global]   [POLL] copyspace1: Triggering collection
[2021-06-24T00:05:02Z INFO  mmtk::scheduler::gc_work] End of GC
[2021-06-24T00:05:02Z INFO  mmtk::util::alloc::large_object_allocator] LargeObjectAllocator.alloc(192012)
[2021-06-24T00:05:03Z INFO  mmtk::plan::global]   [POLL] copyspace0: Triggering collection
[2021-06-24T00:05:03Z INFO  mmtk::scheduler::gc_work] End of GC
[2021-06-24T00:05:03Z INFO  mmtk::plan::global]   [POLL] copyspace1: Triggering collection
[2021-06-24T00:05:03Z INFO  mmtk::scheduler::gc_work] End of GC
[2021-06-24T00:05:03Z INFO  mmtk::plan::global]   [POLL] copyspace0: Triggering collection
[2021-06-24T00:05:03Z INFO  mmtk::scheduler::gc_work] End of GC
[2021-06-24T00:05:03Z INFO  mmtk::plan::global]   [POLL] copyspace1: Triggering collection
[2021-06-24T00:05:03Z INFO  mmtk::scheduler::gc_work] End of GC
[2021-06-24T00:05:04Z INFO  mmtk::plan::global]   [POLL] copyspace0: Triggering collection
[2021-06-24T00:05:04Z INFO  mmtk::scheduler::gc_work] End of GC
[2021-06-24T00:05:04Z INFO  mmtk::plan::global]   [POLL] copyspace1: Triggering collection
[2021-06-24T00:05:04Z INFO  mmtk::scheduler::gc_work] End of GC
[2021-06-24T00:05:05Z INFO  mmtk::plan::global]   [POLL] copyspace0: Triggering collection
[2021-06-24T00:05:05Z INFO  mmtk::scheduler::gc_work] End of GC
===== DaCapo fop PASSED in 2684 msec =====
```

Reply:
I was exactly doing the same thing, but I was missing the `R` at the beginning of the `RBaseBaseSemiSpace` when I tried to rebuild mmtk-jikesrvm.
It works after adding the `R`